### ISSUE TITLE
[BUG] - Table

### ISSUE THREAD

**Issue by keroshanpillay** [nextui-org/nextui#2517]
### NextUI Version

2.0.28

### Describe the bug

The example provided in the async table loading does not type the item pulled from the API. An interface should be defined as 

```
interface Person {
  name: string;
  height: string;
  mass: string;
  birth_year: string;
}

```
then you can type 
`let list = useAsyncList<Person>`

### Steps to Reproduce the Bug or Issue

1. Go to [Infinite Pagination](https://nextui.org/docs/components/table#infinite-pagination)
2. copy the code into a nextjs environment
3. attempt to run

### Expected behavior

The code should have run out-of-the-box (being an example) but it does not. Therefore, we fix

### Operating System Version

maxOS

### Browser

Chrome

**Comment by wingkwong** [comment]
@keroshanpillay Thanks for reporting. However, I'm able to see the type is included in the example locally. Somehow the one in production is outdated. @kuri-sun This issues looks like the one you fixed in #2513. Can you also take a look please?

<img width="749" alt="image" src="https://github.com/nextui-org/nextui/assets/35857179/f0fddc34-270c-498d-b824-448eaa9c11f9">

**Comment by kuri-sun** [comment]
Thanks @wingkwong, you are right. I am going to take a look at this one! 
Also, Thanks @keroshanpillay for sharing the solution as well!
